contracts/LiquidityMiningCampaign.sol

```diff
@@ -0,0 +1,209 @@
+				if(additionalRewards > 0) {
+
+				LockScheme(lockSchemes[i]).updateUserAccruedRewards(_userAddress, additionalRewards);
+				}
+
+				uint256 bonus = LockScheme(lockSchemes[i]).exit(_userAddress);
+				IERC20Detailed(rewarsToken).safeTransfer(_userAddress, bonus);
+				
+			}
+			_exit(_userAddress);
+			userAccruedRewads[_userAddress] = 0;
+
+			emit ExitedAndUnlocked(_userAddress);
+	}
+
+
+	function setReceiverWhitelisted(address receiver, bool whitelisted) override(StakeTransferer) onlyFactory public {
+		StakeTransferer.setReceiverWhitelisted(receiver, whitelisted);
+	}
+
+	// /** @dev exits the current campaign and trasnfers the stake to another whitelisted campaign
+	// 	@param transferTo address of the receiver to transfer the stake to
+	//  */
+	// function exitAndTransfer(address transferTo, address _lockScheme) virtual public onlyWhitelistedReceiver(transferTo) nonReentrant {
+	// 	UserInfo storage user = userInfo[msg.sender];
+		
+	// 	updateRewardMultipliers(); // Update the accumulated multipliers for everyone
+
+	// 	if (user.amountStaked == 0) {
+	// 		return;
+	// 	}
+
+	// 	updateUserAccruedReward(msg.sender); // Update the accrued reward for this specific user
+
+	// 	LockScheme(_lockScheme).exit(msg.sender);
+	// 	claimBonus(_lockScheme, msg.sender);
+	// 	_claim(msg.sender);
+	// 	stakingToken.safeApprove(transferTo, user.amountStaked);
+
+	// 	StakeReceiver(transferTo).delegateStake(msg.sender, user.amountStaked);
+
+	// 	totalStaked = totalStaked.sub(user.amountStaked);
+	// 	user.amountStaked = 0;
+
+	// 	for (uint256 i = 0; i < rewardsTokens.length; i++) {
+	// 		user.tokensOwed[i] = 0;
+	// 		user.rewardDebt[i] = 0;
+	// 	}
+	// }
+
+	function exitAndStake(address _stakePool) public  {
+
+			_exitAndStake(msg.sender, _stakePool);
+	}
+
+	function _exitAndStake(address _userAddress,address _stakePool) public onlyWhitelistedReceiver(_stakePool) {
+			
+		UserInfo storage user = userInfo[_userAddress];
+		
+		if (user.amountStaked == 0) {
+			return;
+		}
+
+		updateRewardMultipliers();
+		updateUserAccruedReward(_userAddress);
+			//todo check how to secure that 0 is the albt
+			uint256 finalRewards = user.tokensOwed[0].sub(userAccruedRewads[_userAddress]);
+			uint256 userBonus;
+			uint256 amountToStake;
+			for (uint256 i = 0; i < lockSchemes.length; i++) {
+
+				uint256 additionalRewards = calculateProportionalRewards(_userAddress, finalRewards, lockSchemes[i]);
+				if(additionalRewards > 0) {
+
+				LockScheme(lockSchemes[i]).updateUserAccruedRewards(_userAddress, additionalRewards);
+				}
+				userBonus = LockScheme(lockSchemes[i]).exit(_userAddress);
+				amountToStake = amountToStake.add(userBonus);
+			}
+
+		 amountToStake = amountToStake.add(user.tokensOwed[0]);
+		_exit(_userAddress);
+
+		IERC20Detailed(rewarsToken).safeApprove(_stakePool, amountToStake);
+		StakeReceiver(_stakePool).delegateStake(_userAddress, amountToStake);
```

Comment:
This is how you print money. Reason for this is that ALBT reward might have been 100, and the bonus 10 more and you would have `amountToStake` equal to 110. However, when you call `_exit`, the initial 100 reward will be transferred back to the user. Then the delegate and stake will stake 110 more and the user ends up with 100 more.